rcl/test/rcl/test_events.cpp

```diff
@@ -59,23 +66,57 @@ class CLASSNAME(TestEventFixture, RMW_IMPLEMENTATION) : public ::testing::Test
     rcl_node_options_t node_options = rcl_node_get_default_options();
     ret = rcl_node_init(this->node_ptr, name, "", this->context_ptr, &node_options);
     ASSERT_EQ(RCL_RET_OK, ret) << rcl_get_error_string().str;
+    ts = ROSIDL_GET_MSG_TYPE_SUPPORT(test_msgs, msg, Primitives);
+  }
+
+  rcl_ret_t SetupPublisher(
+    const rmw_time_t & deadline,
+    const rmw_time_t & lifespan,
+    const rmw_time_t & liveliness_lease_duration,
+    const rmw_qos_liveliness_policy_t liveliness_policy)
+  {
+    // init publisher
+    publisher = rcl_get_zero_initialized_publisher();
+    rcl_publisher_options_t publisher_options = rcl_publisher_get_default_options();
+    publisher_options.qos.deadline = deadline;
+    publisher_options.qos.lifespan = lifespan;
+    publisher_options.qos.liveliness = liveliness_policy;
+    publisher_options.qos.liveliness_lease_duration = liveliness_lease_duration;
+    return rcl_publisher_init(&publisher, this->node_ptr, this->ts, this->topic,
+             &publisher_options);
+  }
+
+  rcl_ret_t SetupSubscriber(
+    const rmw_time_t & deadline,
+    const rmw_time_t & lifespan,
+    const rmw_time_t & liveliness_lease_duration,
+    const rmw_qos_liveliness_policy_t liveliness_policy)
+  {
+    // init publisher
+    subscription = rcl_get_zero_initialized_subscription();
+    rcl_subscription_options_t subscription_options = rcl_subscription_get_default_options();
+    subscription_options.qos.deadline = deadline;
+    subscription_options.qos.lifespan = lifespan;
+    subscription_options.qos.liveliness = liveliness_policy;
+    subscription_options.qos.liveliness_lease_duration = liveliness_lease_duration;
+
+    return rcl_subscription_init(&subscription, this->node_ptr, this->ts, this->topic,
+             &subscription_options);
   }
 
   void SetUpPubSub(
-    const rcl_publisher_event_type_t pub_event_type,
-    const rcl_subscription_event_type_t sub_event_type)
+    const rcl_publisher_event_type_t & pub_event_type,
+    const rcl_subscription_event_type_t & sub_event_type)
   {
     rcl_ret_t ret;
 
-    const char * topic = "rcl_test_publisher_subscription_events";
-    const rosidl_message_type_support_t * ts =
-      ROSIDL_GET_MSG_TYPE_SUPPORT(test_msgs, msg, Primitives);
+    rmw_time_t lifespan {0, 0};
+    rmw_time_t deadline {DEADLINE_PERIOD_IN_MS / 1000, 0};
```

Comment:
I'm not seeing these constants in this file - did they get removed somewhere in a merge confusion? Supposed to be around line 32, right before `#ifdef RMW_IMPLEMENTATION` IIRC.

Also need to set nanos - `ms & THOUSAND * MILLION` - may be worth defining a utility function somewhere `ms_to_rmw_time`

Reply:
They are here as it builds, I've moved that over. Also may be worth adding to rmw then.